{
	"id": "oval:org.debian:def:239392838246311750259714359748631012010",
	"class": "vulnerability",
	"metadata": {
		"title": "CVE-2021-4189 pypy3",
		"affected": {
			"family": "unix",
			"platform": "Debian GNU/Linux 12",
			"product": "pypy3"
		},
		"reference": [
			{
				"source": "CVE",
				"ref_id": "CVE-2021-4189",
				"ref_url": "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-4189"
			}
		],
		"description": "A flaw was found in Python, specifically in the FTP (File Transfer Protocol) client library in PASV (passive) mode. The issue is how the FTP client trusts the host from the PASV response by default. This flaw allows an attacker to set up a malicious FTP server that can trick FTP clients into connecting back to a given IP address and port. This vulnerability could lead to FTP client scanning ports, which otherwise would not have been possible.",
		"debian": {}
	},
	"criteria": {
		"operator": "AND",
		"criterias": [
			{
				"operator": "OR",
				"criterias": [
					{
						"operator": "AND",
						"criterions": [
							{
								"test_ref": "oval:org.debian.oval:tst:2",
								"comment": "all architecture"
							},
							{
								"test_ref": "oval:org.debian.oval:tst:21889",
								"comment": "pypy3 DPKG is earlier than 7.3.5+dfsg-2"
							}
						]
					}
				]
			}
		],
		"criterions": [
			{
				"test_ref": "oval:org.debian.oval:tst:1",
				"comment": "Debian 12 is installed"
			}
		]
	}
}
